GVFS/GVFS.Common/GVFSEnlistmentHealthCalculator.cs

```diff
@@ -0,0 +1,274 @@
+﻿using GVFS.Common.Database;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.Common
+{
+    /// <summary>
+    /// Class responsible for the business logic involved in calculating the health statistics
+    /// of a gvfs enlistment. Constructed with the lists of paths for the enlistment, and then
+    /// internally stores the calculated information. Compute or recompute via CalculateStatistics
+    /// with an optional parameter to only look for paths which are under the specified directory
+    /// </summary>
+    public class GVFSEnlistmentHealthCalculator
+    {
+        // In the context of this project, hydrated files are placeholders or modified paths
+        // The total number of hydrated files is this.PlaceholderCount + this.ModifiedPathsCount
+        private readonly GVFSEnlistmentPathData enlistmentPathData;
+
+        public GVFSEnlistmentHealthCalculator(GVFSEnlistmentPathData pathData)
+        {
+            this.enlistmentPathData = pathData;
+        }
+
+        public GVFSEnlistmentHealthData CalculateStatistics(string parentDirectory, int directoryOutputCount)

```

Comment:
Well since we already calculate all of the directories it would be easy enough to return the entire list, so now that's what happens instead. This means that the client will be responsible for selecting the directories that they want.